ccda author and dates

sjpadgett opened the pull request:
- 3 chances for author
- datetime validation changes

sjpadgett commented:
@adunsulag Sorry, fell asleep!:)
Dates still feels sloppy but still working.
Author is correct for document but still needs component authors looked into

adunsulag commented:
@sjpadgett Thanks Jerry!  I briefly looked over the code and will take a deeper dive tomorrow morning.

adunsulag commented on interface/modules/zend_modules/module/Carecoordination/src/Carecoordination/Model/EncounterccdadispatchTable.php:
```diff
@@ -2224,18 +2234,23 @@ public function getCarecoordinationModuleSettingValue($field_name)
+        JOIN modules AS mo ON mo.mod_directory='Carecoordination'
+        JOIN module_configuration AS conf ON conf.field_value=u.id AND mo.mod_id=conf.module_id
+        WHERE conf.field_name=?";
+        } elseif (is_int($field_name)) {
```
If I'm following this correctly, you treat the field_name as the user id if its an integer?

sjpadgett replied:
yep

sjpadgett commented:
okay just pushed your +1 in author fix.

adunsulag commented on interface/modules/zend_modules/module/Carecoordination/src/Carecoordination/Model/CcdaServiceRequestModelGenerator.php:
```diff
@@ -84,11 +84,11 @@ public function create_data($pid, $encounter, $sections, $components, $recipient
                 <name>" . htmlspecialchars($representedOrganization['name'] ?? '', ENT_QUOTES) . "</name>
-                <telecom use='WP' value='" . htmlspecialchars($representedOrganization['telecom'] ?? '', ENT_QUOTES) . "'/>
-                <streetAddressLine>" . htmlspecialchars($representedOrganization['streetAddressLine'] ?? '', ENT_QUOTES) . "</streetAddressLine>
+                <telecom use='WP' value='" . htmlspecialchars($representedOrganization['phone'] ?? '', ENT_QUOTES) . "'/>
```
I'm guessing this fixes the phone issues we were running into?

sjpadgett replied:
yep, you were looking at translate but this never made it there. I was using the wrong telephone template in header author.

adunsulag replied:
awesome!  I'm glad you found it!

sjpadgett commented:
I'm going to fix the referral then hit bed. Will pickup on dates tomorrow

sjpadgett commented:
@adunsulag where are you seeing the provider req for referral?

adunsulag commented:
@sjpadgett 
SLI Test Script
![image](https://user-images.githubusercontent.com/228117/175756260-118114d6-c87e-49ed-b9fc-b5303cd54614.png)

![image](https://user-images.githubusercontent.com/228117/175756281-658d45ec-de9c-4fac-92dc-08697d2217ce.png)

sjpadgett commented:
okay give me a quick call

adunsulag commented:
@sjpadgett There were a couple of things I had to fix.  I put out a pull request to your repository with the fixes here: https://github.com/sjpadgett/openemr/pull/21

I'm passing Alice with the changes.  Great job on all of the timezone changes!  The scenario MDHT fails when I don't override the time zone because its using the current timezone settigs.  The ETT validator even caught that the timezone was not matching the scenario, when I switched to the 06/22/2015 11:00 AM EST everything passed.   Very pleased with how that turned out so great job.

I gotta run for an hour or so but I'll then go through and verify Jeremy and Happy Kid to make sure that's working on the ETT, then I'll do the visual inspection on the timezones as well to make sure nothing's missing. Looks like phone numbers are working as well (from what I quickly saw).

sjpadgett commented:
added primary diagnosis code to documentationOf
```
<serviceEvent classCode="PCPR">
      <code code="386661006" displayName="Fever Finding" codeSystem="2.16.840.1.113883.6.96" codeSystemName="SNOMED CT"/>
      
```

sjpadgett commented:
add time zone to encounter date
```
<effectiveTime value="202006221000-0400"/>
```

sjpadgett commented:
working on componentOf
```
"primary_diagnosis": {
        "encounter_date": "2020-06-22 10:00:00-0400",
        "encounter_end_date": "2020-06-22 10:30:00-0400",
        "code": "386661006",
        "code_type": "SNOMED CT",
        "text": "Fever (finding)",
        "status": "Completed"
    }
```

sjpadgett commented:
I am just so pleased with this since @adunsulag did his style improvements.

![Summarization-of-Episode-Note](https://user-images.githubusercontent.com/1859985/175816105-82960567-7e01-43a8-b83a-a2b0cb3e524f.png)

sjpadgett commented:
@adunsulag 
Did a lot of work on dates, time zones, author etc. I added to translate time a utc -0400 offset for any date time precision tz, All b1 scenarios pass.
I changed the encounter and note author time to reflect the CCM author time. I believe this may be a mistake. We should discuss tomorrow.

Unsure if you want this merged before you test so, merge if you want or need and I'm not around in the a.m.

adunsulag commented on ccdaservice/oe-blue-button-generate/lib/translate.js:
```diff
@@ -75,12 +75,18 @@ var precisionToFormat = {
-    let result = m.format(formatSpec);
+    if (input.precision === 'tz') {
+        formatSpec = precisionToFormat['tz'];
+        result =  m.utcOffset('-0400').format(formatSpec);
```
When I mean sending it for the entire document, I'm meaning sending it in php the timezone offset as part of the xml (-400) in this case.

sjpadgett replied:
We'd have to bring in the moment-timezone library to set this global. Using environment var and the browsers doesn't seem to give the timezone as they once did. 
We could grab info from PHP and set but for now, this gets us there.

adunsulag replied:
yep, I agree.  I'll do a smoke screen test run against this PR and if everything is looking good I'll bring it in for a full test run today.

adunsulag commented:
@sjpadgett Just did a test run with b1 on the send and everything is passing the validator. Great Job!  I also went through all the XML and the timezone is populated for authors for everything but the lab results.  I'll merge this in and fix the lab results authorship piece so no biggy there. 

I thought when I had read the companion guide that we only want the encompassing encounter on Encounter Summary type documents such as the Referral form.  So we would skip it for CCD since it covers multiple encounters.  I'm basing that on page 37-38 of the Companion Guide: http://www.hl7.org/ccdasearch/pdfs/Companion_Guide.pdf

sjpadgett commented:
I test for multiple encounters see L-2530 of generator `if (isOne(all.encounter_list.encounter) === 1)`. If is one, componentOf shows. Regardless of document type, if documents focus is a single encounter then a componentOf with the encompassingEncounter MAY be included.

adunsulag commented:
ok, sounds good.  I'll remove my docType checks then in my PR.
